opus/application/static_media/js/browse.js

```diff
@@ -696,90 +701,20 @@ var o_browse = {
-            //  load is triggered).
-            let tableCurrentObsNum = obsNum;
-            obsNum = (o_utils.floor((obsNum - 1)/galleryBoundingRect.x) *
-                      galleryBoundingRect.x + 1);
-
-            if (maxSliderVal >= obsNum) {
-                // "sliderObsNum" will be the startObs.
-                // "scrollbarObsNum" will be the obsNum at current scrollbar location.
-                // In table view, it's the top obsNum. In gallery view, it's one of obsNums in the top row.
-                $(`${tab} .op-gallery-view`).infiniteScroll({
-                    "sliderObsNum": obsNum,
-                    "scrollbarObsNum": tableCurrentObsNum
-                });
-                $(`${tab} .op-data-table-view`).infiniteScroll({
-                    "sliderObsNum": obsNum,
-                    "scrollbarObsNum": tableCurrentObsNum
-                });
-                opus.prefs[startObsLabel] = obsNum;
-
-                $(`${tab} .op-observation-number`).html(o_utils.addCommas(obsNum));
-                $(`${tab} .op-slider-pointer`).css("width", `${o_utils.addCommas(maxSliderVal).length*0.7}em`);
-
-                // just make the step size the number of the obserations across the page...
-                // if the observations have not yet been rendered, leave the default, it will get changed later
-                if (galleryBoundingRect.x > 0) {
-                    o_browse.gallerySliderStep = galleryBoundingRect.x;
-                }
-                $("#op-observation-slider").slider({
-                    "value": obsNum,
-                    "step": o_browse.gallerySliderStep,
-                    "max": maxSliderVal,
-                });
-            }
-            $(`${tab} .op-slider-nav`).removeClass("op-button-disabled");
-            // update startobs in url when scrolling
-            o_hash.updateHash(true);
+            // When we update the slider, the following steps are run:
+            // 1. Get the current startObs and top obsNum (top item in table view and one of
+            //    top row items in gallery view) that scrollbar is at.
+            // 2. Get the max value for slider.
+            // 3. Get the scrollbar offset for both gallery and table view.
+            // 4. Update the slider value and data in infiniteScroll instances and URL using
+            //    the data obtained from above 2 steps.
+            let [obsNum, currentScrollObsNum] = o_browse.realignDOMAndGetStartObsAndScrollbarObsNum(selector,
+                                                tab, viewNamespace, galleryBoundingRect, browserResized);
+            let maxSliderVal = o_browse.getSliderMaxValue(viewNamespace, galleryBoundingRect);
+            let [galleryOffset, tableOffset] = o_browse.getScrollbarOffset(tab, obsNum, currentScrollObsNum);
+
+            o_browse.updateSliderValue(tab, galleryBoundingRect, startObsLabel, maxSliderVal,
```

Comment:
was curious why you made this a function when it is only used once and it takes 7 input parameters.  I can see that it shortens the original function, is that the reason?  I'm on the fence about that sort of function creation.

Reply:
I haven't looked at the code carefully here, but I think Debby thought that most of these passed parameters could actually be computed inside the called function since they were only used for that one purpose. That would reduce the number of passed arguments and make it look cleaner.

Reply:
I think that the only thing you should actually pass in is obsNum , currentScrollObsNum, galleryOffset & tableOffset.  The var maxSliderVale is only used by this function and can be retrieved in the function; the others can just be retrieved inside the function as it is not time consuming or cycle-intensive to just get those values again. 
In addition, getScrollbarOffset should return an object vs. a list of parameters <see interesting discussion in slack>. This will avoid potential future issues w/parameter ordering, etc.